lineapy/utils/config.py

```diff
@@ -134,7 +134,6 @@ def set(self, key: str, value: Any, verbose=True) -> None:
                 try:
                     new_db = create_lineadb_engine(value)
                     self.__dict__[key] = value
-                    os.environ[f"LINEAPY_{key.upper()}"] = str(value)
```

Comment:
I don't think we should remove this line. 

The reason why we need to set the environment variable value at the beginning is not obvious in the code (my bad that I didn't comment on this). 

It needs to start with what happens when the `database_url` is changed. When we change the backend database, we need to reset the Linea graph session, too, since `LineaPy` can no longer access the graph in the previous database. If we reset the session, we reset all variable information, and all the configs(we make from CLI and in session) are gone simultaneously. Thus, we need a method to keep all the configuration items outside the python session. And the most straightforward solution I came up with is putting them into environment variables.

Reply:
For more information, this is the line that reset the ipython/jupyter session.
https://github.com/LineaLabs/lineapy/blob/815a9d6e15158ab9b91693a6bc566d6603b34131/lineapy/editors/ipython.py#L126

Reply:
I think the problem its actually not limited to `database_url` because modules are run on import and we recreate options as a variable in this module instead of having a getter.
 https://github.com/LineaLabs/lineapy/blob/2c7249210fa519d1a054096b8fcb043f75d017aa/lineapy/utils/config.py#L237 

Options is recreated from env vars and config.json only ... this means that if a user sets a local change through CLI, which only updates options through a `set` call, that change won't even reflect in other modules which import options later. 

I'm actually surprised we don't have a test that might catch this. Do we have tests using these CLI options? 

Maybe we should make our options acquired through a getter function that accesses a singleton. Then the way we deal with the reset is to save the singleton before the reset, and reassign it after. I don't think we should persist configs set from  the CLI and session that will persist to other CLIs and sessions. Let me play a bit with python today and see what exactly happening to options the way we have it defined here.

Reply:
Let me take a look

Reply:
Upon further investigation, `STATE.ipython.reset()` is not the reason why os.environ.set is required.  

**HOWEVER**, this step is required to transfer lineapy options from the main process to the subprocess running jupyter/ipython which runs in a separate namespace and does not immediately inherit options in the main thread when started from the CLI. 

e.g 
starting `lineapy --logging-level="NOTSET" ipython` and then running `lineapy.options.get("logging-level")` would show the logging level flag being ignored if env variables are not set.

Alternatives to communicate between the main lineapy process and the subprocess running ipython/jupyter were explored, 

- passing the vars through the `subprocess.run` call in cli.
- writing the options to a temp file and loading in the `load_ipython_extension` hook.

Using env variables seems to be the easiest way for now and we will be sticking to it.

This necessitates being able to serialize `storage_options` correctly to string. This PR is now updated to do that correctly. 

@mingjerli